File: Dinner 18_10_2025/ArmStrong.py
number = int(input("Enter a number: "))
initialInput = 0;
initialInput = number;
seperatedNumber = 0;
cubeOfSeperatedNumber = 0;
SumOfCubeOfSeperatedNumber = 0;
while number % 10 !=0 or number / 10 !=0:
	seperatedNumber = number % 10
	cubeOfSeperatedNumber = seperatedNumber**3
	SumOfCubeOfSeperatedNumber += cubeOfSeperatedNumber
	number /=10;
	seperatedNumber = 0;
if initialInput == SumOfCubeOfSeperatedNumber:
	print(f"{initialInput} is an strong number")
else:
	print(f"{initialInput} is not an strong number")

File: Dinner 18_10_2025/ArmStrong.java
import java.util.Scanner;

public class ArmStrong{
	public static void main(String[] args){
		int initialInput = 0;
		Scanner input = new Scanner(System.in);
		System.out.print("Enter a number: ");
		int number = input.nextInt();
		initialInput = number;
		int seperatedNumber = 0;
		int cubeOfSeperatedNumber = 0;
		int SumOfCubeOfSeperatedNumber = 0;

		while(number % 10 !=0 || number / 10 !=0){
			seperatedNumber = number % 10;
			cubeOfSeperatedNumber = seperatedNumber*seperatedNumber*seperatedNumber;
			SumOfCubeOfSeperatedNumber += cubeOfSeperatedNumber;
			number /=10;
			seperatedNumber = 0;
		}
		if (initialInput == SumOfCubeOfSeperatedNumber){
			System.out.printf("%d is an strong number", initialInput);
		} else {
			System.out.printf("%d is not an strong number", initialInput);
		}
	}
}
